src/web_filesystem_watcher.rs

```diff
@@ -0,0 +1,74 @@
+use crossbeam_channel::Receiver;
+use notify::{Config, Event, RecommendedWatcher, RecursiveMode, Result, Watcher};
+use std::path::Path;
+
+#[allow(missing_docs)]
+pub struct FilesystemWatcher {
+    pub watcher: RecommendedWatcher,
+    pub receiver: Receiver<Result<Event>>,
+}
+
+impl Default for FilesystemWatcher {
+    fn default() -> Self {
+        let (sender, receiver) = crossbeam_channel::unbounded();
+        let watcher: RecommendedWatcher = RecommendedWatcher::new(
+            move |res| {
+                sender.send(res).expect("Watch event send failure.");
+            },
+            Config::default(),
+        )
+        .expect("Failed to create filesystem watcher.");
+        FilesystemWatcher { watcher, receiver }
+    }
+}
+
+impl FilesystemWatcher {
```

Comment:
Is this implementation based on the regular bevy filsystem watcher?

Reply:
The FilesystemWatcher struct was completely copied from the default implementation, which is not public.

Reply:
Okay, makes sense :) Maybe just add a comment with a link to the bevy implementation, so we know where to look when updating bevy version?